src/core/dash_internal.h

```diff
@@ -1188,6 +1189,18 @@ auto Segment<Key, Value, Policy>::FindIt(Hash_t key_hash, Pred&& pred) const ->
   return Iterator{};
 }
 
+template <typename Key, typename Value, typename Policy>
+void Segment<Key, Value, Policy>::Prefetch(Hash_t key_hash) const {
+  uint8_t bidx = BucketIndex(key_hash);
+  const Bucket& target = bucket_[bidx];
+  uint8_t nid = NextBid(bidx);
+  const Bucket& probe = bucket_[nid];
+
+  // Prefetch buckets that might hold the key with high probability.
+  __builtin_prefetch(&target, 0, 1);
+  __builtin_prefetch(&probe, 0, 1);
```

Comment:
during high load this can create extra pressure on memory bandwidth

Reply:
but we are going to look up these keys anyways

Reply:
no, I have not. I will check.

Reply:
Could you also check these changes on the 64 or if possible 128 core/thread CPU where memory bandwidth pressure can be higher